archive/brave-core/src/test/java/brave/interop/TracerAdapterTest.java

```diff
@@ -40,7 +44,11 @@
       .tracer();
   Brave brave3 = TracerAdapter.newBrave(brave4);
 
-  @After public void close(){
+  @Before public void clearBrave3State() {
+    ThreadLocalServerClientAndLocalSpanState.clear();
```

Comment:
wild guess is this was the source of flakes. In the brave3 model you could leak contexts because there was no try/finally mechanism.